### DISCUSSION TITLE
Many littles questions

### DISCUSSION THREAD

**Discussion by Pierre-Moulin** [Ifsttar/I-Simpa#324] [Q&A]
(Je suis francais, vous pouvez répondre dans cette langue)

Hello, I have many little questions about the soft and acoustic modeling

-Can we change the colors of the legend 
<60 --> Blue / >80 -->Red Smtk like this and if possible with programable steps

- Could we create a spectrum with Octaves ? 
and if I just declare 8 frequencies (125 --> 8000) should I use an attention of 9 dB = -10 log(8)

-I still having mesh problems and dont know what to do :( 
I have modelized the structure in Solidworks while using only surfaces (which is a bit hard) but always have coplanar surfaces :/ 

-If someone have advices for model an industry room (I have already do the tuto) : 
1) How create the sources ? Should I measure 5 points around my machine and do the average ? I dont have acess to the acoustic power ... I just can do measure in a noisy environment ...
2) Should I model the structure of my  machines ? I don't know why but on an other soft called Rayplus, without it that was pretty correct.
3) What is the purpose of the background noise in I SIMPA ? Its utilization is essential ? 

Thanks for your support :)

**Comment by nicolas-f** [comment]
Hi,

Legend colors can be changed in the settings:
https://i-simpa-wiki.readthedocs.io/fr/latest/menus.html#menu-file
I-Simpa use Gimp palettes file formats located here:
```I-Simpa/Bitmaps/iso```

You can compute using octave only yes. The sound source and attenuation must be prepared accordingly.

1. To get the noise level of a sound source in a noisy environment, this is not ideal but you can do the following calculation:

The principle is to measure with and without the machine running. In the two measurements, the background noise must be the same.

```python
>>> from math import *
>>> background = 73.1
>>> background_with_machine = 74.2
>>> machine_noise = 10*log10(10**(background_with_machine/10.0)-10**(background/10.0))
>>> machine_noise
67.69768640520599
```
2. As you are not aware of the directivity of the noise source, no you should not include the structure of the noise source. As for other objects in the rooms it depends on the size ( acoustic wave length ).
3. background noise is only there for noise indicators calculations, however I don't think we use it.

For 3D models. You should use FreeCad for designing or fixing:
https://grabcad.com/tutorials/freecad-repairing-3d-files-for-3d-printing

best regards,

**Reply by Pierre-Moulin** [reply]
Hello,
I finally clear importing my structure but I always need advises. 
1) As my goal is to model a sound map with/without panels above my machines, is it not indicative without the structure of the machines as the sound go bellow the panels. Do you have an alternative ? 
2) If I want to check the clarity C50/C80 ... what type of source should I use ? 
I tried with a white noise source at the middle but it seems not good 
![image](https://user-images.githubusercontent.com/106664450/186871163-99d471c8-5bac-43b4-868a-e1c916ffd7f2.png)

**Reply by Pierre-Moulin** [reply]
I still don't understand how to set up my legend. Same for come back to the deffault parameters. 
![image](https://user-images.githubusercontent.com/106664450/187153127-2ce01091-b802-4e65-ab7d-7cd8b4d7cc88.png)
